lib/facter/util/wmi.rb

```diff
@@ -2,7 +2,10 @@ module Facter::Util::WMI
   class << self
     def connect(uri = wmi_resource_uri)
       require 'win32ole'
-      WIN32OLE.connect(uri)
+      #WIN32OLE.connect(uri)
+      swbem = WIN32OLE.new("WbemScripting.SWbemLocator")
```

Comment:
This indeed interesting. Looks like OleInitialize calls CoInitializeEx with COINIT_MULTITHREADED (against what the documentation states), so it can be safely ignored. I'll ask the Nano team.

```
require 'ffi'

module Functions
    extend FFI::Library        
    ffi_convention :stdcall

    ffi_lib :ole32
    attach_function :OleInitialize, [:pointer], :int32
    attach_function :CoInitialize, [:pointer], :int32
    attach_function :CoInitializeEx, [:pointer, :int32], :int32    
end  

puts Functions.OleInitialize(FFI::Pointer::NULL)
puts Functions.CoInitialize(FFI::Pointer::NULL)
# COINIT_APARTMENTTHREADED
puts Functions.CoInitializeEx(FFI::Pointer::NULL, 2)
# COINIT_MULTITHREADED
puts Functions.CoInitializeEx(FFI::Pointer::NULL, 0)

```

Output on Nano TP5:

```
0
-2147467263
-2147467263
1
```